extmod/btstack/modbluetooth_btstack.c

```diff
@@ -161,12 +301,17 @@ STATIC void btstack_packet_handler(uint8_t packet_type, uint8_t *packet, uint8_t
         mp_bluetooth_gap_on_connected_disconnected(irq_event, conn_handle, 0xff, addr);
     #if MICROPY_PY_BLUETOOTH_ENABLE_CENTRAL_MODE
     } else if (event_type == GATT_EVENT_QUERY_COMPLETE) {
-        DEBUG_EVENT_printf("  --> gatt query complete\n");
         uint16_t conn_handle = gatt_event_query_complete_get_handle(packet);
         uint16_t status = gatt_event_query_complete_get_att_status(packet);
+        DEBUG_EVENT_printf("  --> gatt query complete irq=%d conn_handle=%d status=%d\n", irq, conn_handle, status);
         if (irq == MP_BLUETOOTH_IRQ_GATTC_READ_DONE || irq == MP_BLUETOOTH_IRQ_GATTC_WRITE_DONE) {
-            // TODO there is no value_handle available to pass here
-            mp_bluetooth_gattc_on_read_write_status(irq, conn_handle, 0, status);
+            // TODO there is no value_handle available to pass here.
+            // TODO try and get this implemented in btstack.
+            mp_bluetooth_gattc_on_read_write_status(irq, conn_handle, 0xffff, status);
+            // Unref the saved buffer for the write operation on this conn_handle.
+            if (irq == MP_BLUETOOTH_IRQ_GATTC_WRITE_DONE) {
+                assert(btstack_finish_pending_operation(MP_BLUETOOTH_BTSTACK_PENDING_WRITE, conn_handle, 0xffff));
```

Comment:
don't put code that must be executed in `assert()`, because it won't be executed with `NDEBUG`